### DISCUSSION TITLE
Trying to view database, getting IDBKey collate message

### DISCUSSION THREAD

**Discussion by BlindSpy** [sqlitebrowser/sqlitebrowser#2667] [Q&A]
Hello everyone, 

I am a less-than-novice Atari/Commodore 64/Sega Genesis/Nintendo generation poor soul that is trying to view the contents of a SQLite database. I'm getting a message saying "A table in this database requires a special collation function 'IDBKEY' that this application can't provide without further knowledge. If you choose to proceed, be aware bad things can happen to your database. Create a backup!"

I'm trying to reverse engineer a total screw up I made by reconstructing (or finding) the keystore file I need to import a wallet.  I have a string that was found in the app .plist file that begins with DBKey and a long string follows.

1. Is this the DBKey the database is looking for?
2. if yes, how do I have the database recognize it and show me the contents?
3. or am I totally off base with not a clue even if the clues were handed to me on a sheet of paper...

Any help would be greatly appreciated. If anyone can help me restore the keystore file (thinking this is where it is), a sizable donation will be made to the charity of your choice.

<img width="480" alt="Screen Shot 2021-04-15 at 10 09 41 PM" src="https://user-images.githubusercontent.com/81504238/114993201-4997b100-9e37-11eb-90e7-9ec02627e026.png">

<img width="166" alt="Screen Shot 2021-04-15 at 10 03 35 PM" src="https://user-images.githubusercontent.com/81504238/114992418-70091c80-9e36-11eb-8b59-48bd5fa77351.png">

Thanks everyone,
Lost-in-the-sauce like a blind spy.

**Comment by MKleusberg** [comment]
Can you provide some more details, like what app were you using for that wallet and where did you find the IndexedDB.sqlite3 file?

In general: The key you are mentioning (the one which is named DBKey) could be the right one. But it's not what DB4S is missing when it gives you the message "A table in this database ...". You can just proceed when you see that message (but especially when you make any changes, you should make a copy of the file) and it will show you all the data in the database in plain text without any further key.

**Reply by BlindSpy** [reply]
Hi MKleusberg, 

The app is VeChainThor Wallet that was made in 2018 and its open source (they are on GitHub). I've attached screen shots that may help.

<img width="843" alt="Screen Shot 2021-05-05 at 6 39 54 PM" src="https://user-images.githubusercontent.com/81504238/117242756-9422a280-add1-11eb-89e0-0931fd29eb36.png">

<img width="833" alt="Screen Shot 2021-05-05 at 6 40 34 PM" src="https://user-images.githubusercontent.com/81504238/117242769-9d137400-add1-11eb-9ce4-fe776766d492.png">

I'm more than happy to reimburse an expert for his/her time in helping me unlock and piece together any information possible to help regain access to the wallet.

Thanks!

**Reply by dbakailash** [reply]
Hello! 
I'm in the same boat, were you able to figure this out?

Thank you!

**Reply by chrisjlocke** [reply]
@bitranarizpe - The collate message can be ignored (of sorts) - it shouldn't affect how you view the data.
